make/Makefile

```diff
@@ -0,0 +1,59 @@
+GO=go
+CMREL=/home/ashley/workspace/cert-manager-release/bin/cmrel
+CTR=docker
```

Comment:
> do we expect for this to make it easy to pass another tool, i.e podman?

That's exactly why I did it yeah; my experience in the past has been that they can be used pretty interchangably as long as you're not doing anything too fancy. For our use cases I think it should be simple enough to change between them (and I did use podman a little while testing this makefile stuff)

> Thought: having used ko for internal project...

I totally accept there could be a better tool for this; I tried not to add any new tools here just to limit the size of the change (and because there's a risk that none of this gets used :wink: )

I do think this approach makes it _very_ easy to build alternative images (as you mention) which is a definite win. Honestly because building with this makefile is very quick, I don't think speed is a great motivator here to use something else.

Reply:
> Do we expect for this to make it easy to pass another tool, i.e podman? I guess the required build args might differ though

I have tested with both `nerdctl` and `podman`; it seems like the subset of commands that we are using are compatible with all three (docker, nerdctl and podman) which is great because I default to using nerdctl in a rootless way :+1: